### DISCUSSION TITLE
where is Toga in OSX Big Sur?

### DISCUSSION THREAD

**Discussion by hoffenmerCode** [beeware/toga#1301] [Q&A] [answered]
Trying to update some toga modules. Running OSX Big Sur. Where in the world is the toga package installed? Tried looking in every location advised online - nothing. 

I need that to update a few fixes from GitHub to the source code. Unless there is another way to do it that I'm not aware of?

**Comment by freakboy3742** [comment, marked as answer]
So - this is an almost impossible question to answer, because it depends entirely on how you installed Python, and how you installed Toga.

If you've got a stock Big Sur machine, and you've installed the Xcode developer tools, and you've installed using `pip install toga`, and you're *not* using a virtual environment, then it's *probably* in `/usr/local/lib/python3.8/site-packages` (possibly a different path depending on the version of Python).

However, there's a lot of "if" in that statement. Every Python installation technique, virtual environment etc changes the answer to your question.

The easiest way to find out for yourself is to open a Python prompt, and run:

    import toga
    toga.__file__

That will give you the literal file that is providing the install of toga that is being used by the Python interpreter in it's current configuration. Start a different interpreter, or a different virtual environment, and the answer might be different.
